File: project/src/main/java/br/com/project/repositories/UserRepository.java
package br.com.project.repositories;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import br.com.project.models.User;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

	public boolean existsByUsername(String username);
	public Optional<User> findByUsername(String username);
	public User findByToken(String token);
}

File: src/main/kotlin/br/com/project/projetoIntegrador/repositories/UserRepository.kt
package br.com.project.projetoIntegrador.repositories

import br.com.project.projetoIntegrador.models.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, Long> {

    fun existsByUsername(username: String): Boolean
    fun findByUsername(username: String): User
    fun findByToken(token: String): Optional<User>
}